src/software/simulated_tests/simulated_er_force_sim_test_fixture.cpp

```diff
@@ -29,6 +30,13 @@ void SimulatedErForceSimTestFixture::SetUp()
 {
     LoggerSingleton::initializeLogger(TbotsGtestMain::logging_dir);
 
+    friendly_mutable_thunderbots_config = std::make_shared<ThunderbotsConfig>();
+    enemy_mutable_thunderbots_config    = std::make_shared<ThunderbotsConfig>();
+    friendly_thunderbots_config = std::const_pointer_cast<const ThunderbotsConfig>(
+        friendly_mutable_thunderbots_config);
+    enemy_thunderbots_config = std::const_pointer_cast<const ThunderbotsConfig>(
+        enemy_mutable_thunderbots_config);
```

Comment:
Curious why we are initializing these variables twice? in `SetUp` and the constructor

Reply:
We need it in here, rather than in the constructor because we want a fresh set of configs between tests

If something registers callbacks on the constructed dynamic parameter instances and goes out of scope, we segfault

So we can potentially remove the ones in the constructor

We need a comment here explaining this above

Reply:
Interesting. From my understanding and [this page](http://google.github.io/googletest/faq.html#CtorVsSetUp), a new test fixture is created for every `TEST_F` (the constructor and `SetUp` are both called), so I'm not exactly sure what difference it would make. But if it fixes your problem then we should keep it.